### PR TITLE
Implements signInWithFacebook function for firebase_auth

### PR DESCRIPTION
**Pull request by frederickcook**
Implements signInWithFacebook function for firebase_auth requiring a Facebook accessToken.

### CODE REVIEW

**Comment by mit-mit** [comment]
Thanks for contributing @frederickcook!

There are some formatting errors flagged by Travis; can you try to run your code though the formatter (run `flutter format .` from the root dir)?

Also, the example app is failing to build with an analysis error:

```
/home/travis/build/flutter/plugins/packages/firebase_auth/android/src/main/java/io/flutter/firebaseauth/FirebaseAuthPlugin.java:120: error: cannot find symbol
    AuthCredential credential = FacebookAuthProvider.getCredential(accessToken);
                                ^
  symbol:   variable FacebookAuthProvider
  location: class FirebaseAuthPlugin
```

**Comment by frederickcook** [comment]
Thanks @mit-mit! Excited to be a contributor!

I ran the formatter and pushed, but have not yet sorted out that build error. I posted [here](https://stackoverflow.com/questions/45336851/firebase-android-facebookauthprovider-cannot-find-symbol) and [here](https://github.com/flutter/flutter/issues/11414) to try to recruit some help.

I realize I probably submitted the PR prematurely so feel free to close and I can re-submit once that is sorted if you prefer.

**Comment by frederickcook** [comment]
@mit-mit I've got this cleaned up - let me know if there are any other comments!

File: packages/firebase_auth/android/src/main/java/io/flutter/firebaseauth/FirebaseAuthPlugin.java [hunk 1]
```diff
@@ -59,6 +60,9 @@ public void onMethodCall(MethodCall call, Result result) {
       case "signInWithGoogle":
         handleSignInWithGoogle(call, result);
         break;
+      case "signInWithFacebook":
+        handleSignInWithGoogle(call, result);
```
Comment (goderbauer) [inline packages/firebase_auth/android/src/main/java/io/flutter/firebaseauth/FirebaseAuthPlugin.java]: Should this call `handleSignInWithFacebook`?

**Comment by goderbauer** [comment]
This is now live: https://pub.dartlang.org/packages/firebase_auth

**Pull request merged**